Pr create https wss endpoints on multiple regions

AndyBoWu opened the pull request.

fxfactorial commented on devops/create_https_wss_endpoints_aws/main.py:
```diff
@@ -0,0 +1,520 @@
+        * dict_region_tgarn
+        *
+    """
+    for i in range(len(ARRAY_OF_REGIONS)):
```
Range on array of regions directly , use enumerate if really need the index

AndyBoWu replied:
could do that too

fxfactorial commented on devops/create_https_wss_endpoints_aws/main.py:
```diff
@@ -0,0 +1,520 @@
+                    )
+                    print("--registering an explorer node into TWO target groups (tg-https and tg-wss) in region " + region)
+                    sleep(4)
+                except Exception as e:
```
Exception is too base a class to just catch blindly on. Either except on specific exceptions like something from boto or remove the try except and let it explode

LeoHChen commented on devops/create_https_wss_endpoints_aws/devnet.json:
```diff
@@ -0,0 +1,43 @@
+  "us-west-2-exp-1": [],
+  "us-west-2-exp-2": [],
+  "us-west-2-exp-3": [],
+  "explorers_0" : ["54.191.139.12", "54.71.27.215"],
```
why two IPs?

what's explorer_0, explorer_1?

AndyBoWu replied:
`explorer_0` has been deprecated. It was created for TG updating. 
There are usually 2 explorers for each endpoint for HA reason.

LeoHChen commented on devops/create_https_wss_endpoints_aws/devnet.json:
```diff
@@ -0,0 +1,43 @@
+  "explorers_0" : ["54.191.139.12", "54.71.27.215"],
+  "explorers_1" : ["54.241.155.180"],
+  "explorers_2" : ["54.91.183.211"],
+  "tg_s0" : ["tg-s0-api-b-https", "tg-s0-api-b-wss"],
```
is it just a name or what? or pre-defined target group?  If it is just a name generated, why add it to the config file?

AndyBoWu replied:
deprecated design

LeoHChen commented on devops/create_https_wss_endpoints_aws/devnet.json:
```diff
@@ -0,0 +1,43 @@
+  "tg_s1" : ["tg-s1-api-b-https", "tg-s1-api-b-wss"],
+  "tg_s2" : ["tg-s2-api-b-https", "tg-s2-api-b-wss"],
+  "tg_s3" : ["tg-s3-api-b-https", "tg-s3-api-b-wss"],
+  "ws_endpoints": ["ws.s0.b.hmny.io", "ws.s1.b.hmny.io", "ws.s2.b.hmny.io","ws.s3.b.hmny.io"]
```
what's the relation between the end points and the above values like tg, explorer?  this json config file has to be structured, it is flat right now and hard to figure out the relation among those values.

LeoHChen commented on devops/create_https_wss_endpoints_aws/main.py:
```diff
@@ -0,0 +1,520 @@
+
+PURPOSE:
+
+create https/wss endpoints for a network
```
Is this scrip always used to create end points from scratch?  Can it be used to update existing end points, such like we launched testnet multiple times? If yes, how?  Please add it in the usage.

AndyBoWu replied:
yes, it is described in the __main__ func.

LeoHChen commented on devops/create_https_wss_endpoints_aws/main.py:
```diff
@@ -0,0 +1,520 @@
+def parse_network_config(param):
+    """ load the network configuration file, retrieve the value by its key """
+    with open(network_config, 'r') as f:
+        network_config_dict = json.load(f)
```
it is easy to convert json to dict in python, so you'd consider a bit more structure data.

AndyBoWu commented on devops/create_https_wss_endpoints_aws/devnet.json:
```diff
@@ -0,0 +1,43 @@
+  "num_of_regions": 4,
+  "regions": ["us-east-2", "us-west-1"],
+  "region_vpc": ["vpc-2c420a44","vpc-bb770fdc"],
+  "regions_prod": ["us-east-1","us-east-2","us-west-1"],
```
`regions` is the region we would like to create endpoints on. 

`regions_prod` is a reference, cuz it matches the vpc id in the region. Usually, we won't touch this key-value pair.

AndyBoWu commented:
changes on feb 1, 2020

1. removed NUM_OF_REGIONS from json file and main.py, redundant
2. renamed `regions_prod` to `regions_ref`; renamed `region_vpc_prod` to `region_vpc_ref`
3. removed `regions_code` from json, deprecated
4. code refactoring for the ssl cert requesting
